Code_Exercises/Exercise_7_Unified_Shared_Memory_Ext/solution.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ SYCL Academy (c)
+
+ SYCL Academy is licensed under a Creative Commons
+ Attribution-ShareAlike 4.0 International License.
+
+ You should have received a copy of the license along with this
+ work.  If not, see <http://creativecommons.org/licenses/by-sa/4.0/>.
+*/
+
+#define SYCL_ACADEMY_USING_COMPUTECPP
+
+#define CATCH_CONFIG_MAIN
+#include <catch2/catch.hpp>
+
+#ifdef SYCL_ACADEMY_USING_COMPUTECPP
+#include <SYCL/experimental/usm_wrapper.h>
+#endif  // SYCL_ACADEMY_USING_COMPUTECPP
+
+#include <CL/sycl.hpp>
+
+#ifdef SYCL_ACADEMY_USING_COMPUTECPP
+#include <SYCL/experimental.hpp>
+#endif  // SYCL_ACADEMY_USING_COMPUTECPP
+
+#include <numeric>
+
+struct usm_device_selector : public cl::sycl::device_selector {
+  int operator()(const cl::sycl::device& d) const override {
+    if (d.get_info<sycl::info::device::usm_device_allocations>()) {
+      return 1;
+    }
+    else {
+      return -1;
+    }
+  }
+};
+
+template <typename T>
+class add;
+
+template <typename T>
+void parallel_add(std::vector<T> &inputA, std::vector<T> &inputB,
+                  std::vector<T> &output) {
+  using namespace cl::sycl;
+
+#ifdef SYCL_ACADEMY_USING_COMPUTECPP
+  using namespace cl::sycl::experimental;
+#endif  // SYCL_ACADEMY_USING_COMPUTECPP
+
+  assert(inputA.size() == inputB.size() && inputA.size() == output.size());
```

Comment:
```suggestion
  assert((inputA.size() == inputB.size()) && (inputA.size() == output.size()));
```

Also you should include `cassert`.

Reply:
Fixed manually.